packages/reactivity/src/ref.ts

```diff
@@ -193,13 +308,72 @@ class CustomRefImpl<T> {
   }
 }
 
+/**
+ * Creates a customized ref with explicit control over its dependency tracking
+ * and updates triggering.
+ *
+ * It expects a factory function, which receives `track` and `trigger` functions
+ * as arguments and should return an object with `get` and `set`.
+ *
+ * @example
+ * Using a custom ref to implement debounce with `v-model`:
+ * ```js
+ * // in the template: <input v-model="text" />
+ *
+ * function useDebouncedRef(value, delay = 200) {
+ *   let timeout
+ *   return customRef((track, trigger) => {
+ *     return {
+ *       get() {
+ *         track()
+ *         return value
+ *       },
+ *       set(newValue) {
+ *         clearTimeout(timeout)
+ *         timeout = setTimeout(() => {
+ *           value = newValue
+ *           trigger()
+ *         }, delay)
+ *       }
+ *     }
+ *   })
+ * }
+ *
+ * export default {
+ *   setup() {
+ *     return {
+ *       text: useDebouncedRef('hello')
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * @param factory The function that receives the `track` and `trigger` callbacks.
+ */
 export function customRef<T>(factory: CustomRefFactory<T>): Ref<T> {
   return new CustomRefImpl(factory) as any
 }
 
 export type ToRefs<T = any> = {
   [K in keyof T]: ToRef<T[K]>
 }
+
+/**
+ * Creates an object of refs pointing to the given reactive object's properties.
```

Comment:
Let's reuse the api version for this one https://vuejs.org/api/reactivity-utilities.html#torefs